### DISCUSSION TITLE
Replaying Events

### DISCUSSION THREAD

**Discussion by FinalFortune** [rasmus/EventFlow#881] [Q&A]
I would like to replay my events until a certain point in time, after reading through the code and 1 or 2 other posts I came to the following solution:

1. Use `IReadModelPopulator `to purge all the read models
2. Use `IEventStore` to load and select events
3. Apply these events using `IReadStoreManager`

Is this all that is necessary to set the program state to a certain point in time? what about the aggregate store, would I need to change that somehow?

My second problem is while my services use a single _EventStore_ service to track their aggregates, each service is almost entirely isolated and register only the domain classes necessary to the respective service.  So when I call `LoadAllEvents` on the `IEventStore` it tries to deserialize unknown events, but I don't see a way to load the relative events without loading all then filtering them.

Any workarounds for this? Perhaps it's best to add a method to IEventStore which will not throw an exception or not deserialize events that aren't registered?

**Comment by rasmus** [comment]
Hi @FinalFortune if you domains are isolated, moving them to separate persistence stores seems like a good solution or did you figure something else.
